pkg/controller/volume/persistentvolume/controller_base.go

```diff
@@ -509,6 +509,29 @@ func (ctrl *PersistentVolumeController) upgradeVolumeFrom1_2(volume *api.Persist
 	return true
 }
 
+// findDeleterParameters finds StorageClass.Parameters of storage class where
+// given volume belongs
+func (ctrl *PersistentVolumeController) findDeleterParameters(volume *api.PersistentVolume) (map[string]string, error) {
+	className := getVolumeClass(volume)
+	if className == "" {
+		return nil, nil
+	}
+
+	obj, found, err := ctrl.classes.GetByKey(className)
```

Comment:
My opinion is that the PV should have all the properties it needs to be cleaned up.  You could optionally propagate changes to StorageClass properties to PVs that were provisioned by that StorageClass but this may only be desirable in certain scenarios. 

Consider a NFS server is provisioning from its own local storage and it fills up and a new nfs-server starts provisioning the same class of storage.  You want users to continue to use the same name of storageClass, but the provisioned backing volumes have different nfs host servers.   If NFS exports required cleanup you'd always be cleaning up the newest server from the storageClass config and not the server that had the export.